sdks/java/core/src/main/java/org/apache/beam/sdk/io/DatastoreIO.java

```diff
@@ -387,6 +388,19 @@ public long getEstimatedSizeBytes(PipelineOptions options) throws Exception {
     }
 
     @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+      builder
+          .addIfNotDefault("host", host, DEFAULT_HOST)
+          .addIfNotNull("dataset", datasetId)
+          .addIfNotNull("namespace", namespace);
+
+      if (query != null) {
+        builder.add("query", query.toString());
```

Comment:
We handle a small set of explicit types for display data (string, integer, floating point, java class, duration, timestamp, boolean), and we have `.add` overloads for the corresponding java types.

We chose not to add a catch-all for `Object` so that specifying the type for display data is deliberate. This adds some protection against inadvertently using the wrong type (`MyCustomTimestamp`), or a type which doesn't implement toString (`MyCustomStruct`).